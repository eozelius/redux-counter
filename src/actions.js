// Action types
export const INCREMENT = 'INCREMENT'
export const DECREMENT = 'DECREMENT'

// Action Creators
export const incrementCounter = () => {
  return {
    type: INCREMENT
  }
}

export const decrementCounter = () => {
  return {
    type: DECREMENT
  }
}
