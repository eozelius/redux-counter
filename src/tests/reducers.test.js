import { INCREMENT, DECREMENT, incrementCounter, decrementCounter } from '../actions.js'
import counterReducer from '../reducers'

describe('redux Counter Actions', () => {
  it('defaults to the initial state', () => {
    const action = { type: 'dummy_action' }
    const initialState = { count: 0 }
    expect(counterReducer(undefined, action)).toEqual(initialState)
  })

  it('increments the count', () => {
    const initialState = { count: 0 }
    const action = incrementCounter()
    const expectedState = {
      count: 1
    }
    expect(counterReducer(initialState, action)).toEqual(expectedState)
    expect(counterReducer(initialState, action)).toMatchSnapshot()
  })

  it('decrements the count', () => {
    const initialState = { count: 0 }
    const action = decrementCounter()
    const expectedState = {
      count: -1
    }
    expect(counterReducer(initialState, action)).toEqual(expectedState)
    expect(counterReducer(initialState, action)).toMatchSnapshot()
  })

})
