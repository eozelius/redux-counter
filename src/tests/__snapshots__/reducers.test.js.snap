// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redux Counter Actions decrements the count 1`] = `
Object {
  "count": -2,
}
`;

exports[`redux Counter Actions increments the count 1`] = `
Object {
  "count": 2,
}
`;
